server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -585,8 +586,27 @@ private static boolean parsesArrayValue(Mapper mapper) {
         return mapper instanceof FieldMapper && ((FieldMapper) mapper).parsesArrayValue();
     }
 
-    private static void parseNonDynamicArray(DocumentParserContext context, final String lastFieldName, String arrayFieldName)
-        throws IOException {
+    private static void parseNonDynamicArray(
```

Comment:
Maybe also add unit tests to `DocumentParserTests` that test with store array source? Including some tests with multi levels of object array fields and store array source param being enabled on different levels.

Reply:
I added tests in `IgnoredSourceFieldMapperTests`, along with the rest of the tests for extended synthetic source. Any suggestions for scenarios to cover?

Reply:
I missed that the tests in `IgnoredSourceFieldMapperTests` also invoke this method. The tests `IgnoredSourceFieldMapperTests` look good to me.